### DISCUSSION TITLE
OR query for relationship table

### DISCUSSION THREAD

**Discussion by SherifMega** [PostgREST/postgrest#3828] [Q&A] [answered]
### i need to know how to apply (_**or like**_) for 2 columns in relationship

i'm fetching `jobs` with it's `customers` i need to search by **or like** within `first_name` and `last_name` of `customers`

```js
  let query = supabase
    .from("jobs")
    .select(
      `
        *, 
        custFilter:customers!inner(),
        customers:customers(first_name, last_name)
      `,
      { count: "exact" }
    )
    .range(offset, offset + pageSize - 1);

  // Add search filter
  if (searchQuery) {
    const isNumber = !isNaN(Number(searchQuery));
    if (isNumber) {
      //this is because like operator not working with integer columns
      query = query.filter("id", "eq", Number(searchQuery));
    } else {
      query = query
      .or(`custFilter.first_name.ilike%${searchQuery}%,custFilter.last_name.ilike%${searchQuery}%`)
      .or(
        `custFilter.not.is.null`
      );
    }
  }
```

error
```json
{
    "code": "PGRST100",
    "details": "unexpected \"f\" expecting \"not\" or operator (eq, gt, ...)",
    "hint": null,
    "message": "\"failed to parse logic tree ((custFilter.first_name.ilike%rob%,custFilter.last_name.ilike%rob%))\" (line 1, column 15)"
}
```

**Comment by SherifMega** [comment]
@laurenceisla pls can you support ?

**Reply by laurenceisla** [reply]
[Wolfgang's response below](https://github.com/PostgREST/postgrest/discussions/3828#discussioncomment-11638919) tackles everything you need to make the query work.

**Reply by laurenceisla** [reply]
Just to add, in supabase-js follow [the example given for "Use `or` on referenced tables"](https://supabase.com/docs/reference/javascript/or).

**Reply by SherifMega** [reply]
thanks ❤️

**Comment by wolfgangwalther** [comment, marked as answer]
> `custFilter.first_name.ilike%rob%`

Two problems with that:
- % is problematic because of percent encoding in URLs. use `*` instead as [mentioned in the docs](https://docs.postgrest.org/en/v12/references/api/tables_views.html#operators).
- once inside `or` you need to replace `=` with `.`. See [examples in the docs](https://docs.postgrest.org/en/v12/references/api/tables_views.html#logical-operators).

Thus: `custFilter.first_name.ilike.*rob*` should work better.

**Reply by wolfgangwalther** [reply]
Ah, and a third problem: or with embedding doesn't work like that.

See the [docs](https://docs.postgrest.org/en/v12/references/api/resource_embedding.html#embedded-filters).

Instead of `or=(custFilter.first_name.ilike.*rob*,...` it needs to be something along the lines of `custFilter.or=(first_name.ilike.*rob*,...)`.

**Reply by SherifMega** [reply]
thanks a lot ❤️ it works fine 
this is updated typescript code based on docs you need to pass  `{ foreignTable: "custFilter" }` as separated params for or

```ts
  let query = supabase
    .from("jobs")
    .select(
      `
        *,
        custFilter:customers(first_name,last_name),
        customers:customers(first_name, last_name)
      `,
      { count: "exact" },
    )
    .range(offset, offset + pageSize - 1);

  // Add search filter
  if (searchQuery) {
    const isNumber = !isNaN(Number(searchQuery));
    if (isNumber) {
      query = query.or(`id.eq.${Number(searchQuery)}`);
    } else {
      query = query.or(
        `first_name.ilike.*${searchQuery}*,last_name.ilike.*${searchQuery}*`,
        { foreignTable: "custFilter" },
      ).or('custFilter.not.is.null');
    }
  }

```
